accounts/models.py

```diff
@@ -0,0 +1,108 @@
+"""
+Account models
+"""
+from django.contrib.auth.base_user import AbstractBaseUser
+from django.db import models
+
+from django.contrib.auth.models import UserManager
+from django.db.models.signals import post_save
+from django.utils import timezone
+
+
+class User(AbstractBaseUser):
+    """
+    User model
+    """
+    role = models.CharField(max_length=12,
+                            default='employee',
+                            error_messages={
+                                'required': "Role must be provided"
+                            })
+    email = models.EmailField(unique=True, blank=False,
+                              error_messages={
+                                  'unique': "A user with that email already exists.",
+                              })
+    is_email_confirmed = models.BooleanField(default=False, blank=False)
+    is_active = models.BooleanField(default=True, blank=False)
+
+    USERNAME_FIELD = "email"
+
+    objects = UserManager()
+
+    class Meta:
+        db_table = 'user'  # define custom table name
+
+
+class EmployeeProfile(models.Model):
+    """
+    Employee profile model
+    """
+    user = models.OneToOneField(User, on_delete=models.CASCADE)
```

Comment:
`first_name` and `last_name` are already available in the `AbstractUser` model

Reply:
AbstractUser includes full user model fields , for example we do not need first_name and last_name in case of employee/company registration instead we need company_name.